main/lsp/requests/go_to_implementation.cc

```diff
@@ -0,0 +1,152 @@
+
+    if (queryResult.error) {
+        // An error happened while setting up the query.
+        response->error = move(queryResult.error);
+        return response;
+    }
+
+    if (queryResult.responses.empty()) {
+        return response;
+    }
+
+    vector<unique_ptr<Location>> result;
+    auto queryResponse = move(queryResult.responses[0]);
+    if (auto def = queryResponse->isDefinition()) {
+        // User called "Go to Implementation" from the abstract function definition
+        core::SymbolRef method = def->symbol;
+        if (!method.data(gs)->isMethod())
+            response->error = make_unique<ResponseError>(
+                (int)LSPErrorCodes::InvalidParams,
+                "Go to implementation can be used only for methods or references of abstract classes");
+
+        core::SymbolRef overridedMethod = method;
+        if (method.data(gs)->isOverride()) {
+            overridedMethod = findOverridedMethod(gs, method);
+        }
+        auto locationsOrError = findMethodImplementations(gs, overridedMethod);
+
+        if (locationsOrError.error != nullptr) {
+            response->error = move(locationsOrError.error);
+            return response;
+        } else {
+            for (const auto &location : locationsOrError.locations) {
+                addLocIfExists(gs, result, location);
+            }
+        }
+    } else if (auto constant = queryResponse->isConstant()) {
+        // User called "Go to Implementation" from the abstract class reference
+        auto classSymbol = constant->symbol;
+
+        if (!classSymbol.data(gs)->isClassOrModule() || !classSymbol.data(gs)->isClassOrModuleAbstract()) {
+            response->error = make_unique<ResponseError>(
+                (int)LSPErrorCodes::InvalidParams,
+                "Go to implementation can be used only for methods or references of abstract classes");
+            return response;
+        }
+
+        auto classOrModuleRef = classSymbol.asClassOrModuleRef();
+        auto childClasses = classOrModuleRef.getSubclasses(gs, false);
+        for (const auto &childClass : childClasses) {
+            for (auto loc : childClass.data(gs)->locs()) {
+                addLocIfExists(gs, result, loc);
+            }
+        }
+
+    } else if (auto send = queryResponse->isSend()) {
+        auto mainResponse = move(send->dispatchResult->main);
+
+        // User called "Go to Implementation" from the abstract function call
+        const core::MethodRef calledMethod = mainResponse.method;
+
+        if (mainResponse.errors.size() != 0) {
+            response->error = makeInvalidParamsError("Failed to fetch implementations");
+            return response;
+        }
+
+        core::SymbolRef overridedMethod = core::SymbolRef(calledMethod);
+        if (calledMethod.data(gs)->isOverride()) {
+            overridedMethod = findOverridedMethod(gs, overridedMethod);
+        }
```

Comment:
```suggestion
        // User called "Go to Implementation" from the abstract function call
        if (mainResponse.errors.size() != 0) {
            response->error = makeInvalidParamsError("Failed to fetch implementations");
            return response;
        }

        auto calledMethod = mainResponse.method;
        core::SymbolRef overridedMethod = calledMethod;
        if (calledMethod.data(gs)->isOverride()) {
            overridedMethod = findOverridedMethod(gs, overridedMethod);
        }
```
The constructor for `SymbolRef` that takes a `MethodRef` argument is not marked explicit, so you don't need be explicit when assigning it to a `core::SymbolRef`. However, if you do want to be explicit, I would recommend something like `core::SymbolRef overridedMethod{calledMethod};` instead.